code/modules/power/supermatter/supermatter_effects.dm

```diff
@@ -0,0 +1,103 @@
+/datum/component/supermatter_glow
+/// The current light colour. Default is the nice yellow!
+	var/light_color = SUPERMATTER_COLOUR
+/// Typecasted reference to our supermatter(parent)
+	var/obj/machinery/power/supermatter_crystal/our_supermatter
+/// Just in case a bug gets introduced later on, let's not crash people!
+	var/loop_started = FALSE
+
+/datum/component/supermatter_glow/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_SUPERMATTER_PROCESS_ATMOS, .proc/update_effects) // When we are added, we listen for this signal and send the proc.
+	our_supermatter = parent // typecast our parent
```

Comment:
Don't like the typecasted var for parent pattern. You only need it typecasted in two procs, so just typecast them in those procs